server.go

```diff
@@ -323,9 +323,21 @@ func (a *App) BatchHandler(w http.ResponseWriter, r *http.Request) {
 		}
 
 		// Object is not found
-		meta, err = a.metaStore.Put(object)
-		if err == nil {
-			responseObjects = append(responseObjects, a.Represent(object, meta, meta.Existing, true, useTus))
+		if bv.Operation == "upload" {
+			meta, err = a.metaStore.Put(object)
+			if err == nil {
+				responseObjects = append(responseObjects, a.Represent(object, meta, false, true, useTus))
```

Comment:
And here, we don't use `meta.Existing` because we know that either (a) the database entry did not previously exist or (b) it did, but the file didn't exist on disk and all we had was a stub in the database.  In neither case is the entry valid for download, since there's no data to back it.